packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx

```diff
@@ -287,6 +352,16 @@ function ChartsXAxis(inProps: ChartsXAxisProps) {
     y: positionSign * (axisHeight - labelHeight),
   };
 
+  /* If there's an axis title, the tick labels have less space to render  */
+  const tickLabelsMaxHeight = Math.max(
+    0,
+    axisHeight - (label ? labelHeight + AXIS_LABEL_TICK_LABEL_GAP : 0) - tickSize - TICK_LABEL_GAP,
+  );
+
+  const tickLabels = isClient
+    ? shortenLabels(visibleLabels, drawingArea, tickLabelsMaxHeight, axisTickLabelProps.style)
+    : new Map();
```

Comment:
Not sure if we should no labels or all labels when server-side rendering.

All labels would make more sense for the cases when there's no overlap, but they wouldn't look good for cases with an overlap 🤔